lib/Dredd.js

```diff
@@ -1,21 +1,89 @@
 const async = require('async');
```

Comment:
The code of this file [still seems ugly to me](https://twitter.com/ismonkeyuser/status/1112958510252412928) at many places and I blame mainly the asynchronous parts. I hope with async/await/promises it could get even more readable and unit-testable in the future.